sdk/python/feast/infra/online_stores/online_store.py

```diff
@@ -258,6 +258,18 @@ async def get_online_features_async(
                 requested_features=requested_features,
             )
 
+            return idxs, read_rows
+
+        all_responses = await asyncio.gather(
+            *[
+                query_table(table, requested_features)
+                for table, requested_features in grouped_refs
+            ]
+        )
+
+        for (idxs, read_rows), (table, requested_features) in zip(
+            all_responses, grouped_refs
+        ):
```

Comment:
when requesting features across multiple tables, we can parallelize the calls to each.